### PR TITLE
Liquid staking - non epoch version

### PR DESCRIPTION
**Pull request by junkai121**
close https://github.com/iqlusioninc/liquidity-staking-module/issues/8

### CODE REVIEW

File: x/staking/types/msg.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ var (
 	_ sdk.Msg                            = &MsgDelegate{}
 	_ sdk.Msg                            = &MsgUndelegate{}
 	_ sdk.Msg                            = &MsgBeginRedelegate{}
+	_ sdk.Msg                            = &MsgTransferTokenizeShareRecord{}
```
Comment (chaindbkey) [inline x/staking/types/msg.go:29]: Missing msg: MsgTokenizeShares, MsgRedeemTokensforShares

File: x/staking/keeper/msg_server.go [hunk 2]
```diff
@@ -373,3 +377,193 @@ func (k msgServer) Undelegate(goCtx context.Context, msg *types.MsgUndelegate) (
...
+	}
+
+	if validator.IsBonded() {
+		k.bondedTokensToNotBonded(ctx, returnAmount)
+	}
+
+	// create reward ownership record
+	k.AddTokenizeShareRecord(ctx, record)
+
+	_, err = k.Keeper.Delegate(ctx, record.GetModuleAddress(), msg.Amount.Amount, sdkstaking.Unbonded, validator, false)
```
Comment (zmanian) [inline x/staking/keeper/msg_server.go:467]: if the type of `acc` is a vesting account, we need to call `TrackDelegation` similar to here
https://github.com/cosmos/cosmos-sdk/blob/8b78406bd3941fc0b98683bac06de55898859486/x/bank/keeper/keeper.go#L476-L477

File: x/staking/keeper/msg_server.go [hunk 3]
```diff
@@ -373,3 +377,193 @@ func (k msgServer) Undelegate(goCtx context.Context, msg *types.MsgUndelegate) (
...
+	if err != nil {
+		return nil, err
+	}
+	err = k.bankKeeper.BurnCoins(ctx, types.NotBondedPoolName, sdk.Coins{msg.Amount})
+	if err != nil {
+		return nil, err
+	}
+
+	// convert the share tokens to delegated status
+	_, err = k.Keeper.Delegate(ctx, delegatorAddress, msg.Amount.Amount, sdkstaking.Unbonded, validator, false)
```
Comment (zmanian) [inline x/staking/keeper/msg_server.go:534]: We need to call TrackDelegation is the account is of the vesting Type here.

**Comment by junkai121** [comment]
TrackDelegation and TrackUndelegation is called internally by UndelegateCoinsFromModuleToAccount and DelegateCoinsFromAccountToModule - added notes here https://github.com/iqlusioninc/liquidity-staking-module/pull/7/commits/dbfab8d0233f4d5b9b9bbe34d2e2700912384339 @zmanian

**Pull request merged**